src/cli/killswitch.rs

```diff
@@ -0,0 +1,76 @@
+use std::path::Path;
+
+const KILLSWITCH_FILE_PATH: &str = "/tmp/lightswitch/killswitch";
+
+pub struct KillSwitch {
+    killswitch_path: String,
+    ignore_killswitch: bool,
+}
+
+impl KillSwitch {
+    pub fn new(killswitch_path_override: Option<String>, ignore_killswitch: bool) -> Self {
+        let killswitch_path = match killswitch_path_override {
+            Some(path) if !path.is_empty() => path,
+            _ => KILLSWITCH_FILE_PATH.to_string(),
+        };
+        KillSwitch {
+            killswitch_path,
+            ignore_killswitch,
+        }
+    }
+
+    pub fn enabled(&self) -> bool {
+        !self.ignore_killswitch && Path::new(&self.killswitch_path).try_exists().unwrap()
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use std::fs::{create_dir_all, remove_file, File};
+
+    #[test]
+    fn test_killswitch_enabled() {
+        // Given
+        create_dir_all("/tmp/lightswitch").unwrap();
+        let killswitch = KillSwitch::new(None, /*ignore_killswitch*/ false);
```

Comment:
nit: there's no linter that keeps these comments synced to the source code so they might get out of date. They should not be necessary as most people in the Rust community use [Rust Analyzer](https://rust-analyzer.github.io/) in their editor that shows the argument names in the call sites 
```suggestion
        let killswitch = KillSwitch::new(None, false);
```

Reply:
Fair enough. Will update this.